### PR TITLE
fix viewer's animation error when animations has same name

### PR DESCRIPTION
**Pull request by cpunion**

### CODE REVIEW

**Comment by bjsplat** [comment]
Please make sure to label your PR with "bug", "new feature" or "breaking change" label(s).
To prevent this PR from going to the changelog marked it with the "skip changelog" label.

**Comment by bjsplat** [comment]
Snapshot stored with reference name:
refs/pull/14247/merge

Test environment: 
https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14247/merge/index.html

To test a playground add it to the URL, for example:

https://babylonsnapshots.z22.web.core.windows.net/refs/pull/14247/merge/index.html#WGZLGJ#4600

Links to test babylon tools with this snapshot:

https://playground.babylonjs.com/?snapshot=refs/pull/14247/merge
https://sandbox.babylonjs.com/?snapshot=refs/pull/14247/merge
https://gui.babylonjs.com/?snapshot=refs/pull/14247/merge
https://nme.babylonjs.com/?snapshot=refs/pull/14247/merge

To test the snapshot in the playground with a playground ID add it after the snapshot query string:

https://playground.babylonjs.com/?snapshot=refs/pull/14247/merge#BCU1XR#0

**Review by RaananW — APPROVED** [review]
An acceptable solution. It means, however, that the second animation with the same name will never play.

**Review by sebavan — DISMISSED** [review]
The comment on the line just before is not uptodate in this case.

Is there no way to allow all of them to play ???

**Comment by cpunion** [comment]
> The comment on the line just before is not uptodate in this case.
> 
> Is there no way to allow all of them to play ???

I agree that we need a better solution. This PR is just to avoid a break. However, I'm not certain about the range that needs to be changed.

**Comment by RaananW** [comment]
This will require changing the value added to the select. it should be the index of the animation instead of the animation name.

**Comment by RaananW** [comment]
Do we want to fix it in this PR?

File: packages/tools/viewer/src/model/viewerModel.ts [hunk 1]
```diff
@@ -393,7 +393,7 @@ export class ViewerModel implements IDisposable {
         // can't use .find, noe available on IE
         const filtered = this._animations.filter((a) => a.name === name.trim());
         // what the next line means - if two animations have the same name, they will not be returned!
```
Comment (RaananW) [inline packages/tools/viewer/src/model/viewerModel.ts:395]:
```suggestion
```

**Pull request merged**